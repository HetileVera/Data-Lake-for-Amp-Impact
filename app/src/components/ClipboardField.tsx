import React from "react";
import { observer } from "mobx-react";
import { Box, Button, HStack, Tooltip, useClipboard } from "@chakra-ui/react";
import { CheckCircleIcon, CopyIcon } from "@chakra-ui/icons";
import isEmpty from "lodash/isEmpty";

import { useColorScheme } from "models/useColorScheme";

interface IClipboardField {
  value: string;
  isPassword?: boolean;
  canCopy?: boolean;
}

export const ClipboardField = observer(({ value, isPassword = false, canCopy = true }: IClipboardField) => {
  const [show, setShow] = React.useState(false);
  const handleShowClick = () => setShow((previous) => !previous);
  const { tone, colorScheme } = useColorScheme();
  const { hasCopied, onCopy } = useClipboard(value, 500);
  const display = isPassword && !show ? "**************************************" : value;

  return (
    <HStack color={tone(700)}>
      <Box p={0.3} pl={0} pr={2} display="inline-block">
        {display}
      </Box>

      {canCopy && (
        <Tooltip label={hasCopied ? "Copied" : "Copy"}>
          {hasCopied ? <CheckCircleIcon data-testid="check-icon" /> : <CopyIcon cursor="pointer" onClick={onCopy} data-testid="copy-icon" />}
        </Tooltip>
      )}

      {isPassword && (
        <Button
          size="xs"
          onClick={handleShowClick}
          colorScheme={colorScheme}
          variant="outline"
          _focus={{ boxShadow: "none" }}
          isDisabled={isEmpty(value)}
        >
          {show ? "Hide" : "Show"}
        </Button>
      )}
    </HStack>
  );
});
