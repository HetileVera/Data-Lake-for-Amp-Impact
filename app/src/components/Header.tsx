import React from "react";
import { observer } from "mobx-react";
import { Box, Stack, Heading, HStack, Image, VStack, StackDivider, Container, useTheme } from "@chakra-ui/react";

// import { theme } from "themes/orange";

import AWSLogo from "images/AWS_logo_RGB_BLK.png";
import SalesforceLogo from "images/Salesforce_Corporate_Logo_RGB.png";
import VeraLogo from "images/vera-solutions-squarelogo-1455866080574.png";

export const Header = observer(() => {
  const theme = useTheme();

  return (
    <Container maxW="container.md" pt="20px" position="relative">
      <VStack spacing="24px">
        <HStack
          direction={{ base: "column", lg: "row" }}
          spacing={{ base: "0", lg: "0" }}
          divider={<StackDivider borderColor="gray.200" />}
          width="100%"
        >
            <Box maxW="90px" paddingRight="24px">
            <Image src={VeraLogo} />
          </Box>
          <Box maxW="90px" paddingRight="24px">
            <Image src={SalesforceLogo} />
          </Box>
          <Box maxW="65px">
            <Image src={AWSLogo} marginLeft="24px" />
          </Box>
        </HStack>
        <Stack direction={{ base: "column", lg: "row" }} spacing={{ base: "0", lg: "0" }} width="100%">
          <Box maxH="160px" marginBottom="32px">
            <Heading id="h2-title" display="inline-block" size="lg" pb="10px" bgGradient={theme.gradients.bgDark} bgClip="text" letterSpacing="-1px">
              Data Lake for Nonprofits, Powered by AWS
            </Heading>
          </Box>
        </Stack>
      </VStack>
    </Container>
  );
});
