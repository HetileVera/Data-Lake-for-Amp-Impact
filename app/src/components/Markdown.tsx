import React from "react";
import { Heading, Link } from "@chakra-ui/react";
import { ExternalLinkIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";
import ReactMarkdown from "react-markdown";
import remarkGfm from "remark-gfm";

interface IMarkdown {
  content: string;
}

export const Markdown = observer(({ content }: IMarkdown) => {
  return (
    <ReactMarkdown
      remarkPlugins={[remarkGfm]}
      components={{
        a: ({ ...props }) => (
          <Link href={props.href} isExternal color="red.600">
            {props.children} <ExternalLinkIcon mx="2px" />
          </Link>
        ),
        li: ({ ...props }) => <li style={{ marginLeft: "20px" }}>{props.children}</li>,
        h1: ({ ...props }) => (
          <Heading as="h1" size="lg" mt={4} mb={2}>
            {props.children}
          </Heading>
        ),
        h2: ({ ...props }) => (
          <Heading as="h2" size="md" mt={4} mb={2}>
            {props.children}
          </Heading>
        ),
        h3: ({ ...props }) => (
          <Heading as="h3" size="sm" mt={4} mb={2}>
            {props.children}
          </Heading>
        ),
      }}
    >
      {content}
    </ReactMarkdown>
  );
});
