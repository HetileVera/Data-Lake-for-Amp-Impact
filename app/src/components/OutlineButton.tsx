import React from "react";
import { Button, ButtonProps, useTheme } from "@chakra-ui/react";

/**
 * A Button with an outline that uses the color ranges that we need
 */
interface IOutlineButton extends ButtonProps {
  selected?: boolean;
}

export const OutlineButton = ({ selected = false, ...props }: IOutlineButton) => {
  const theme = useTheme();
  const outline = theme.outlineButton;

  const allProps: IOutlineButton = {
    colorScheme: outline.scheme,
    size: "md",
    variant: "outline",
    _hover: outline.hover,
    _active: outline.active,
    _focus: { boxShadow: "none" },
    ...props,
  };

  if (selected) {
    allProps.bg = outline.selected.bg;
    allProps.color = outline.selected.color;
  }

  return <Button {...allProps} />;
};
