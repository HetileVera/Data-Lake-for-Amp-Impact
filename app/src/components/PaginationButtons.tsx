// Deprecated
import React from "react";
import { Box, Flex, IconButton, Text } from "@chakra-ui/react";
import { ChevronLeftIcon, ChevronRightIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";

import { useColorScheme } from "models/useColorScheme";

interface IPaginationButtons {
  totalPages: number;
  currentPage: number;
  isProcessing?: boolean;
  setCurrentPage: (page: number) => void;
}

export const PaginationButtons = observer(({ totalPages, currentPage, setCurrentPage, isProcessing = false }: IPaginationButtons) => {
  const { tone, colorScheme } = useColorScheme();

  if (totalPages <= 1) return <Box mr={1} w="210px" mt={1} />;

  const hasNext = currentPage < totalPages;
  const hasPrevious = currentPage > 1;
  const color = isProcessing ? tone(300) : tone(800);

  return (
    <Flex fontSize="xs" justifyContent="flex-end" mr={1} w="210px" mt={1}>
      {(hasPrevious && (
        <IconButton
          _focus={{ boxShadow: "none" }}
          aria-label="previous page"
          size="xs"
          variant="ghost"
          colorScheme={colorScheme}
          icon={<ChevronLeftIcon fontSize="xl" />}
          onClick={() => setCurrentPage(currentPage - 1)}
          pb={1}
          isDisabled={isProcessing}
        />
      )) || <Box w="24px" h="24px" />}

      <Text color={color}>
        Page {currentPage} of {totalPages}
      </Text>

      {(hasNext && (
        <IconButton
          _focus={{ boxShadow: "none" }}
          aria-label="next page"
          size="xs"
          variant="ghost"
          colorScheme={colorScheme}
          icon={<ChevronRightIcon fontSize="xl" />}
          onClick={() => setCurrentPage(currentPage + 1)}
          pb={1}
          isDisabled={isProcessing}
        />
      )) || <Box w="24px" h="24px" />}
    </Flex>
  );
});
