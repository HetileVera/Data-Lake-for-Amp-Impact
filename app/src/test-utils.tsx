import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";

import { theme } from "themes/orange";

export const CustomChakraProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={theme}>
      <div data-testid="test-root">{children}</div>
    </ChakraProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "wrapper">) => render(ui, { wrapper: CustomChakraProvider, ...options });

export * from "@testing-library/react";

export { customRender as render };
