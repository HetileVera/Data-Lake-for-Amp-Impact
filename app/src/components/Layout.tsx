import React from "react";
import { observer } from "mobx-react";
import { Box, Container, Heading, Stack, useTheme } from "@chakra-ui/react";

import { useColorScheme } from "models/useColorScheme";

import { CurvedBox } from "./CurvedBox";
import { Header } from "./Header";
import { StepsIndicator } from "./StepsIndicator";

interface ILayout {
  step: number;
  title: string;
  description: string;
  explanation?: string;
  warning?: string;
  children: React.ReactNode;
}

export const Layout = observer(({ step, title, description, explanation, warning, children }: ILayout) => {
  const theme = useTheme();
  const { tone } = useColorScheme();

  return (
    <Box id={`step${step}`}>
      <CurvedBox bgGradient={theme.gradients.bgLight} />

      <Header />

      <Box position="relative">
        <CurvedBox />
        <Box bg={tone(400)} w="full" position="relative">
          <Container maxW="container.md" pt="15px" pb="15px">
            <StepsIndicator current={step} />
          </Container>
        </Box>
      </Box>

      <Container maxW="container.md" pt="0px" position="relative">
        <Heading display="inline-block" size="lg" pt="16px" pb="30px" color={tone(50)} letterSpacing="-1px">
          {title}
        </Heading>

        <Box color={tone(50)} mb={4}>
          <Stack direction="row" spacing="8px" align="left" justifyContent="space-between">
            <Box mb={6}>
              <Box>{description}</Box>
              {explanation && <Box sx={{ mt: 2 }}>{explanation}</Box>}
              {warning && <Box sx={{ mt: 2 }}>{warning}</Box>}
            </Box>
          </Stack>
        </Box>

        {children}
      </Container>
    </Box>
  );
});
