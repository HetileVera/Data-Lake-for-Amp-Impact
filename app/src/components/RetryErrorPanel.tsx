import React from "react";
import isEmpty from "lodash/isEmpty";
import { Alert, AlertIcon, Box, Button, Flex } from "@chakra-ui/react";
import { observer } from "mobx-react";

import { Markdown } from "components/Markdown";

interface IRetryErrorPanel {
  errorMessage: string;
  errorDetail?: string;
  onRetry: () => void;
}

export const RetryErrorPanel = observer(({ errorMessage, errorDetail, onRetry }: IRetryErrorPanel) => {
  const [expanded, setExpanded] = React.useState(false);
  const hasTechDetail = !isEmpty(errorDetail);

  return (
    <Alert status="error" mt={0} mb={6} color="red.700" alignItems="flex-start" borderRadius="lg">
      <AlertIcon alignSelf="flex-start" />
      <Box>
        <Box>
          {errorMessage}
          {hasTechDetail && (
            <>
              &nbsp;If you continue to get this error, you can peek at the technical details and seek assistance from a solution implementor agency.
            </>
          )}
        </Box>
        <Flex w="full" mt={4} justifyContent="space-between" textAlign="right">
          {hasTechDetail && (
            <Button size="sm" color="red.800" variant="link" onClick={() => setExpanded((previous) => !previous)} _focus={{ boxShadow: "none" }}>
              Click to see technical details
            </Button>
          )}
          {!hasTechDetail && <Box />}
          <Button colorScheme="red" size="sm" loadingText="Processing" onClick={() => onRetry()}>
            Try Again
          </Button>
        </Flex>
        {expanded && errorDetail && (
          <Box mt={6} mb={4}>
            <Markdown content={errorDetail} />
          </Box>
        )}
      </Box>
    </Alert>
  );
});
