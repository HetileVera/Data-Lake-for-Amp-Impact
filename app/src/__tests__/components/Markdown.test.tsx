import { cleanup, screen } from "@testing-library/react";
import { create } from "react-test-renderer";

import { render, CustomChakraProvider } from "test-utils";

import { Markdown } from "components/Markdown";

describe("Markdown component", () => {
  afterEach(cleanup);

  test("snapshot", () => {
    const tree = create(
      <CustomChakraProvider>
        <Markdown content="test-content" />
      </CustomChakraProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("render properly", () => {
    render(<Markdown content="test-content" />);

    expect(screen.getByText("test-content")).toBeInTheDocument();
  });
});
