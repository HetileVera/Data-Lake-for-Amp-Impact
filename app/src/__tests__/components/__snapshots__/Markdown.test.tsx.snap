// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown component snapshot 1`] = `
Array [
  <div
    data-testid="test-root"
  >
    test-content
  </div>,
  <span />,
]
`;
